kernel/os/include/os/arch/pic32/os/os_arch.h

```diff
@@ -38,7 +38,8 @@ typedef uint32_t os_sr_t;
 /* Stack type, aligned to a 32-bit word. */
 #define OS_STACK_PATTERN    (0xdeadbeef)
 
-typedef uint32_t os_stack_t;
+/* uint64_t in an attempt to get the stack 8 aligned */
+typedef uint64_t os_stack_t;
```

Comment:
The latter option would be unbearable :)
I can't think of a thing which would break with the change you're proposing. And if they do, we'll fix them. Maybe supplement it with a comment saying that these lines should not be reordered from this on? So that folks later coming here later would know to be aware what os_arch_task_stack_init() will do.

As for changing the arguments to os_arch_task_stack_init(), I'd be cool with that. Simpler is better.
It would be good to send an email to mailing list regarding that change, though. This because it is a bit of an API change, though I imagine most of the folks wouldn't even notice. Maybe it would make sense to do that via a separate pull request? So that we can get this stack alignment issue fixed sooner rather than later.